### PR TITLE
pulselistener: Use batch_checkout instead of robust_checkout

### PR DESCRIPTION
**Pull request by La0**

### CODE REVIEW

File: lib/cli_common/cli_common/mercurial.py [hunk 1]
```diff
@@ -22,14 +38,40 @@ def robust_checkout(repo_url, repo_dir, branch=b'tip', revision=None):
...
+                                stream=True)
+    hg_run(cmd)
+    log.info('Inital clone finished')
+
+    repo = hglib.open(repo_dir)
+    target = int(repo.identify(rev=revision, num=True).strip().decode('utf-8'))
+    log.info('Target revision for incremental checkout', revision=target)
+
+    steps = list(range(1, target, batch_size)) + [target]
+    for rev in steps:
```
Comment (marco-c) [inline lib/cli_common/cli_common/mercurial.py]: Nit: shouldn't something like `for rev in range(1, target + 1, batch_size)` work?

File: lib/cli_common/cli_common/mercurial.py [hunk 2]
```diff
@@ -5,8 +5,24 @@
 
 import hglib
 
+import cli_common.log
+
+log = cli_common.log.get_logger(__name__)
+
 MOZILLA_CENTRAL = 'https://hg.mozilla.org/mozilla-central'
-MOZILLA_UNIFIED = 'https://hg.mozilla.org/mozilla-unified',
+MOZILLA_UNIFIED = 'https://hg.mozilla.org/mozilla-unified'
```
Comment (marco-c) [inline lib/cli_common/cli_common/mercurial.py]: Nit: are we actually using these two?

File: lib/cli_common/cli_common/mercurial.py [hunk 3]
```diff
@@ -5,8 +5,24 @@
...
+def hg_run(cmd):
+    '''
+    Run a mercurial command without an hglib instance
+    Useful for initial custom clones
+    '''
+    proc = hglib.util.popen([hglib.HGPATH] + cmd)
+    out, err = proc.communicate()
+    if proc.returncode:
+        raise hglib.error.CommandError(cmd, proc.returncode, out, err)
+    return out
```
Comment (calixteman) [inline lib/cli_common/cli_common/mercurial.py]: I wonder if you should kill or not proc when useless

File: lib/cli_common/cli_common/mercurial.py [hunk 4]
```diff
@@ -22,14 +38,40 @@ def robust_checkout(repo_url, repo_dir, branch=b'tip', revision=None):
...
+    log.info('Batch checkout', url=repo_url, dir=repo_dir)
+    cmd = hglib.util.cmdbuilder('clone',
+                                repo_url,
+                                repo_dir,
+                                noupdate=True,
+                                stream=True)
+    hg_run(cmd)
+    log.info('Inital clone finished')
+
+    repo = hglib.open(repo_dir)
```
Comment (calixteman) [inline lib/cli_common/cli_common/mercurial.py]: nit: with hglib.open(...)

File: lib/cli_common/cli_common/mercurial.py (hunk 1 above)
Comment (La0) [reply lib/cli_common/cli_common/mercurial.py]: Not in every case i'm afraid

```
>>> list(range(1, 1000+1, 100))
[1, 101, 201, 301, 401, 501, 601, 701, 801, 901]
```

File: lib/cli_common/cli_common/mercurial.py (hunk 2 above)
Comment (La0) [reply lib/cli_common/cli_common/mercurial.py]: Nope. I'll remove them

File: lib/cli_common/cli_common/mercurial.py [hunk 5]
```diff
@@ -5,8 +5,21 @@
...
+    Useful for initial custom clones
+    '''
+    proc = hglib.util.popen([hglib.HGPATH] + cmd)
+    out, err = proc.communicate()
+    if proc.returncode:
+        raise hglib.error.CommandError(cmd, proc.returncode, out, err)
+    return out
 
 
 def robust_checkout(repo_url, repo_dir, branch=b'tip', revision=None):
```
Comment (marco-c) [inline lib/cli_common/cli_common/mercurial.py]: BTW, we should probably use this everywhere (in SA, CCOV and bugbug)

Comment (La0) [reply lib/cli_common/cli_common/mercurial.py]: robust or batch ?

Comment (marco-c) [reply lib/cli_common/cli_common/mercurial.py]: I meant robust, it seems like we are duplicating this code in all projects

Comment (marco-c) [reply lib/cli_common/cli_common/mercurial.py]: Well, all projects except pulselistener :P

**Pull request merged**